src/ConvertToSARIF.Engine/DiagnosticRecord.cs

```diff
@@ -31,6 +50,28 @@ public DiagnosticRecord(PSObject inputObject)
             }
 
             Message = inputObject?.Properties?[nameof(Message)]?.Value?.ToString();
+
+            dynamic SuppressionList = inputObject?.Properties?[nameof(Suppression)]?.Value;
```

Comment:
You can simply try casting directly to the property. 

`Suppresions = inputObject?.Properties?[nameof(Suppression)]?.Value as List<Suppression>`

But only if you know the type provided by PSScriptAnalyzer. Also using as will simply make the `Suppression` property null in case it doesn't exists or it can't be cast.

I am currently uncertain what type PSScriptAnalyzer will actually return.

Reply:
PSScriptAnalyzer will return two types of records: DiagnosticRecord and SuppressedRecord.

 if I used `Suppresions = inputObject?.Properties?[nameof(Suppression)]?.Value as List<Suppression>`
`Suppresions` will be null.

![image](https://user-images.githubusercontent.com/86460533/127292292-a88f7e73-e776-4350-a2c8-04a1d100c095.png)